pkg/cache/cache_test.go

```diff
@@ -198,6 +198,19 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					}
 				})
 
+				It("should be able to list objects with GVK populated", func() {
+					By("listing pods")
+					out := &kcorev1.PodList{}
+					Expect(informerCache.List(context.Background(), out)).To(Succeed())
+
+					By("verifying that the returned pods have GVK populated")
+					Expect(out.Items).NotTo(BeEmpty())
+					Expect(out.Items).Should(SatisfyAny(HaveLen(3), HaveLen(4)))
```

Comment:
If we are getting 4 pods, I think that is a bug in the namespaced cache implementation. I will attempt to re-create locally.

I think we may need to file an issue and merge this with a TODO remove `HaveLen(4)` IMO. 

/cc @DirectXMan12

Reply:
Don't know if you managed to repro locally or not, but to me it looks like the [regular cache](https://github.com/kubernetes-sigs/controller-runtime/blob/7f16f2b744937bb10e5a3acca56aa5b54f534813/pkg/cache/cache_test.go#L76) will be returning [4 pods](https://github.com/kubernetes-sigs/controller-runtime/blob/7f16f2b744937bb10e5a3acca56aa5b54f534813/pkg/cache/cache_test.go#L100-L103) as it lists across all namespaces(?) whereas the [multinamespaced cache](https://github.com/kubernetes-sigs/controller-runtime/blob/7f16f2b744937bb10e5a3acca56aa5b54f534813/pkg/cache/cache_test.go#L79) will only list 3 as there is a [pod in a namespace](https://github.com/kubernetes-sigs/controller-runtime/blob/7f16f2b744937bb10e5a3acca56aa5b54f534813/pkg/cache/cache_test.go#L103) that it isn't caching?

I haven't looked into this thoroughly though so am kinda guessing how the two different caches behave 🤷‍♂️